kbfsgit/runner.go

```diff
@@ -774,10 +983,18 @@ func (r *runner) processCommands(ctx context.Context) (err error) {
 		cmdParts := strings.Fields(cmd)
 		if len(cmdParts) == 0 {
 			if len(fetchBatch) > 0 {
-				r.log.CDebugf(ctx, "Processing fetch batch")
-				err = r.handleFetchBatch(ctx, fetchBatch)
-				if err != nil {
-					return err
+				if r.cloning {
+					r.log.CDebugf(ctx, "Processing clone")
+					err = r.handleClone(ctx)
+					if err != nil {
+						return err
+					}
+				} else {
+					r.log.CDebugf(ctx, "Processing fetch batch")
+					err = r.handleFetchBatch(ctx, fetchBatch)
```

Comment:
Is the plan to do the copy thing with fetch as well in the future?

Reply:
I don't think it makes sense for regular fetches, because in the common case you'll end up processing way more objects that you need to.  I've done other optimizations for `fetch` in #1159.